src/utils/polymer.dom.html

```diff
@@ -244,7 +244,7 @@
       let list = [];
       for (let i=0, l=c$.length, c; (i<l) && (c=c$[i]); i++) {
         if ((c.nodeType === Node.ELEMENT_NODE) &&
-            Polymer.Utils.matchesSelector(this.node, selector)) {
+            Polymer.Utils.matchesSelector(c, selector)) {
```

Comment:
I believe we shouldn't need to check `.root` there unless there is a Shady DOM bug. Let's deal with that separately if so.